sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -576,6 +601,13 @@ int coap_get_block1(coap_pkt_t *pkt, coap_block1_t *block1)
     return (block1->more >= 0);
 }
 
+int coap_get_block2(coap_pkt_t *pkt, coap_block1_t *block2)
+{
+    block2->more = coap_get_blockopt(pkt, COAP_OPT_BLOCK2, &block2->blknum,
+            &block2->szx);
+    return block2->more;
```

Comment:
The return value must indicate if the option is present, as with coap_get_block1(). See the fix in [f466972](https://github.com/bergzand/RIOT/pull/4/commits/f4669724c8c04503db48055a3eddb45016d7a95e).

Reply:
Ah, must have mixed up the API functions here